### DISCUSSION TITLE
Custom method for URL encoding

### DISCUSSION THREAD

**Discussion by eXenon** [sqlalchemy/sqlalchemy#10713] [Usage Questions] [answered]
Hi!

I have some issues using the output of `str(url)` where `url` is a variable containing an instance of `sqlalchemy.engine.url.URL`.

I noticed that this is due to the method sqlalchemy uses to quote its output:
```
def _sqla_url_quote(text: str) -> str:
    return re.sub(r"[:@/]", lambda m: "%%%X" % ord(m.group(0)), text)
```

[Taken from url.py#L909](https://github.com/sqlalchemy/sqlalchemy/blob/82690b1cfc1e76e5deb622a9afefbcf3be299962/lib/sqlalchemy/engine/url.py#L909)

The unquoting is done via the `unquote` method of `urllib`, but the quoting isn’t. There are other parts of the same file that already use `quote_plus`, so I am intrigued why there is a difference here. Is there a reason for that? Would it be ok for me to submit a PR changing that method to:
```
_sqla_url_quote = quote_plus
```

Thanks so much for the help :) !

Edit: I just noticed that newer versions of sqlalchemy deprecated the use of `__str__` in favor of the more explicit `render_as_string`. My question still stands, although I’ll update my code.

**Comment by CaselIT** [comment, marked as answer]
Hi,

I think this is the same issue as reported here https://github.com/sqlalchemy/sqlalchemy/issues/10662

So sure feel free to open a PR for it

**Reply by eXenon** [reply]
Excellent, thanks! I’ll try and do that.
